src/index.ts

```diff
@@ -1,115 +1,138 @@
-import fetch from 'node-fetch';
-import fs from 'fs';
-import program from 'commander';
-import path from 'path';
+import fetch from "node-fetch";
+import fs from "fs";
+import program from "commander";
+import path from "path";
 
-program.version('0.1.0');
+program.version("0.1.0");
 
 program
-    .requiredOption('--url <url>', 'json url')
-    .option('--output <output>', 'directory name for output images')
-    .option('--map <map>', 'file name for map json file')
+  .requiredOption("--url <url>", "json url")
+  .option("--output <output>", "directory name for output images")
+  .option("--map <map>", "file name for map json file")
+  .option("--source <source>", "file name to save url json source");
 
 interface JsonObject {
-    [key: string]: JsonObject | string
+  [key: string]: JsonObject | string;
 }
 
 const fetchJsonFromUrl = async (url: string): Promise<JsonObject> => {
-    const response = await fetch(url);
-    return await response.json();
-}
+  const response = await fetch(url);
+  return await response.json();
+};
 
 // random string identifier
-const getRandomString = () => Math.random().toString(36).substr(2, 9)
+const getRandomString = () =>
+  Math.random()
+    .toString(36)
+    .substr(2, 9);
 
 // given a json object and result array, collect all jpg/jpeg/png/gif image url in result array
-const collectImageUrls = (jsonObject: JsonObject | string, resultArray: string[]) => {
-    if (typeof jsonObject === 'string') {
-        if (jsonObject.startsWith('https://') &&
-            (jsonObject.endsWith('.jpg') ||
-                jsonObject.endsWith('.jpeg') ||
-                jsonObject.endsWith('.png') ||
-                jsonObject.endsWith('.gif'))
-        ) {
-            resultArray.push(jsonObject)
-        }
-    } else { // recursively traverse
-        Object.values(jsonObject).forEach(node => {
-            collectImageUrls(node, resultArray)
-        });
+const collectImageUrls = (
+  jsonObject: JsonObject | string,
+  resultArray: string[]
+) => {
+  if (typeof jsonObject === "string") {
+    if (
```

Comment:
Stylistically its better to && instead of doing two ifs

Reply:
Consider just making a constant which is like
const isImage = jsonObject.endsWith(".jpg") ||
        jsonObject.endsWith(".jpeg") ||
        jsonObject.endsWith(".png") ||
        jsonObject.endsWith(".gif")